numcodecs/checksum32.py

```diff
@@ -7,30 +7,28 @@
 
 
 from .abc import Codec
-from .compat import ndarray_from_buffer, buffer_copy
+from .compat import ensure_contiguous_ndarray, memory_copy
 
 
 class Checksum32(Codec):
 
     checksum = None
 
     def encode(self, buf):
-        if isinstance(buf, np.ndarray) and buf.dtype == object:
-            raise ValueError('cannot encode object array')
-        arr = ndarray_from_buffer(buf, dtype='u1')
+        arr = ensure_contiguous_ndarray(buf).view('u1')
         checksum = self.checksum(arr) & 0xffffffff
         enc = np.empty(arr.nbytes + 4, dtype='u1')
         enc[:4].view('<u4')[0] = checksum
         enc[4:] = arr
```

Comment:
Should we use `memory_copy` here?

Reply:
Possibly could do, I don't know if would would be any faster.

Reply:
Was mainly thinking about this from a consistency standpoint.